statusHandler/p2pQuota/p2pQuotaProcessor_test.go

```diff
@@ -0,0 +1,241 @@
+package p2pQuota_test
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/statusHandler"
+	"github.com/ElrondNetwork/elrond-go/statusHandler/mock"
+	"github.com/ElrondNetwork/elrond-go/statusHandler/p2pQuota"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewP2pQuotaProcessor_NilStatusHandlerShouldErr(t *testing.T) {
+	t.Parallel()
+
+	pqp, err := p2pQuota.NewP2pQuotaProcessor(nil)
+	assert.True(t, check.IfNil(pqp))
+	assert.Equal(t, statusHandler.ErrNilAppStatusHandler, err)
+}
+
+func TestNewP2pQuotaProcessor_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	pqp, err := p2pQuota.NewP2pQuotaProcessor(&mock.AppStatusHandlerStub{})
+	assert.False(t, check.IfNil(pqp))
+	assert.Nil(t, err)
+}
+
+//------- AddQuota
+
+func TestP2pQuotaProcessor_AddQuotaShouldWork(t *testing.T) {
+	t.Parallel()
+
+	pqp, _ := p2pQuota.NewP2pQuotaProcessor(&mock.AppStatusHandlerStub{})
+	nonExistingIdentifier := "non existing identifier"
+	identifier := "identifier"
+	numReceived := uint32(1)
+	sizeReceived := uint64(2)
+	numProcessed := uint32(3)
+	sizeProcessed := uint64(4)
+
+	pqp.AddQuota(identifier, numReceived, sizeReceived, numProcessed, sizeProcessed)
+
+	nonExistentQuota := pqp.GetQuota(nonExistingIdentifier)
+	assert.Nil(t, nonExistentQuota)
+
+	quota := pqp.GetQuota(identifier)
+	assert.Equal(t, numReceived, quota.NumReceived())
+	assert.Equal(t, sizeReceived, quota.SizeReceived())
+	assert.Equal(t, numProcessed, quota.NumProcessed())
+	assert.Equal(t, sizeProcessed, quota.SizeProcessed())
+}
+
+//------- ResetStatistics
+
+func TestP2pQuotaProcessor_ResetStatisticsShouldEmptyStatsAndCallSetOnAllMetrics(t *testing.T) {
+	t.Parallel()
+
+	identifier1 := "identifier"
```

Comment:
you can remove the `1` from the variables' names

Reply:
agree, done